src/edu/stanford/nlp/tagger/maxent/TestSentence.java

```diff
@@ -821,28 +761,30 @@ public double scoreOf(int[] sequence) {
     return getScores(history);
   }
 
-  // todo [cdm 2013]: Tagging could be sped up quite a bit here if we cached int arrays of tags by index, not Strings
+  // todo [cdm 2013]: Tagging could be sped up quite a bit here if we cached int
+  // arrays of tags by index, not Strings
   protected String[] stringTagsAt(int pos) {
-    if ((pos < leftWindow()) || (pos >= size + leftWindow())) {
+    pos -= leftWindow(); // Adjust once, for everything below:
+    if ((pos < 0) || (pos >= size)) {
       return naTagArr;
     }
 
-    String[] arr1;
-    if (originalTags != null && originalTags.get(pos - leftWindow()) != null) {
-      arr1 = new String[1];
-      arr1[0] = originalTags.get(pos - leftWindow());
-      return arr1;
+    if (originalTags != null) {
+      final String orig = originalTags.get(pos);
+      if (orig != null) {
+        return new String[] { orig };
+      }
     }
 
+    String[] arr1;
     String word = sent.get(pos - leftWindow());
```

Comment:
missed a leftWindow() here :/

Reply:
well spotted. I may have introduced this just yesterday with the rebases, because this fragment had been independently improved (I recall having had a compile error because of this arr1).